### ISSUE TITLE
eslint constantly compains about no-use-before-define rule

### ISSUE THREAD

**Issue by sandy081** [Microsoft/vscode-eslint#248]
From @oliversalzburg  - https://github.com/Microsoft/vscode/issues/27827

- VSCode Version: 1.12.2
- OS Version: Windows 10

Steps to Reproduce:

1. Open [spodr](https://github.com/fairmanager/spodr) in VS Code

ESLint shows an alert saying: `d:\spodr\.eslintrc`: Configuration for rule "no-use-before-define" is invalid: Value "[object Object]" no (or more than one) schemas match.

The rule is defined as:
```json
		"no-use-before-define": [
			2,
			{
				"functions": false,
				"classes": true,
				"variables": true
			}
		],
```

When I run eslint manually from the CLI, I get no errors and behavior in general is as expected.

**Comment by dbaeumer** [comment]
@oliversalzburg I can't reproduce this. I did:

- cloned https://github.com/fairmanager/spodr
- ran npm install
- opened the repository in VS Code
- opened JS from lib and bin

I only received in error from bin/spodr.js

Am I missing something ?

**Comment by oliversalzburg** [comment]
Huh, I don't get it. I had this all the time last week. Now I can't repro it either. I'll watch out if it happens again and report back.

**Comment by dbaeumer** [comment]
@oliversalzburg thanks. I will close the issue for now. Please ping if you see it again.

**Comment by dloret** [comment]
Hi, I had the same issue and discovered that, for me the root cause was that I updated the node version to 8.3.0 using NVM. Unfortunately, I didn't see the 'default' alias was still pointing to an older version, that I uninstalled just after updating. Without seeing it, I was trying to use a 'N/A' node version, which messed with the functioning os eslint.
I just had to point the NVM 'default' alias to the last version of node and make sure I was using it:
